p11-kit/rpc-server.c

```diff
@@ -2086,124 +2087,200 @@ p11_kit_remote_serve_token (CK_FUNCTION_LIST *module,
 	return ret;
 }
 
+static void
+token_set_write_protected (CK_TOKEN_INFO *token, P11KitUri *uri)
+{
+	const char *write_protected =
+		p11_kit_uri_get_vendor_query (uri, "write-protected");
+	if (write_protected &&
+	    _p11_conf_parse_boolean (write_protected, false))
+		token->flags |= CKF_WRITE_PROTECTED;
+	else
+		token->flags &= ~CKF_WRITE_PROTECTED;
```

Comment:
I'm assuming that token->flags and CFK_WRITE_PROTECTED are the same here...

Reply:
Sorry I don't get it; the `else` clause is to turn off the flag in `token->flags`.

Reply:
I don't think integer promotion is possible here; as `token->flags` is defined as `CK_ULONG` and `CKF_WRITE_PROTECTED` is defined as `1UL << 1`.